spacialgaze-plugins/customavatar.js

```diff
@@ -11,29 +11,32 @@
 
 const fs = require('fs');
 const path = require('path');
-const request = require('request');
-
+let Stream = require('stream').Transform;
 // The path where custom avatars are stored.
 const AVATAR_PATH = path.join(__dirname, '../config/avatars/');
 
+let Request;
+
 // The valid file extensions allowed.
 const VALID_EXTENSIONS = ['.jpg', '.png', '.gif'];
 
-function downloadImage(image_url, name, extension) {
-	request
-		.get(image_url)
-		.on('error', err => {
-			console.error(err);
-		})
-		.on('response', response => {
-			if (response.statusCode !== 200) return;
-			let type = response.headers['content-type'].split('/');
-			if (type[0] !== 'image') return;
-
-			response.pipe(fs.createWriteStream(AVATAR_PATH + name + extension));
+function downloadImage(image_url, name) {
+	let ext = path.extname(image_url);
+	if (image_url.startsWith('https://')) Request = require('https');
+	Request = require('http');
```

Comment:
I don't think you understood what @Mystifi said, this will require http everytime since it overwrites the https require on line 26, use an `else` statement to fix that